lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,320 @@
+use crate::prelude::*;
+
+use crate::animation::{Animator, IntervalCounter, linear_interpolation};
+use crate::traits::HasPosition;
+
+use nalgebra::Vector3;
+use nalgebra::zero;
+
+// ====================
+// === PhysicsForce ===
+// ====================
+
+pub trait PhysicsForce {
+    fn force(&self, kinematics:&KinematicsProperties) -> Vector3<f32>;
+}
+
+// ======================
+// === DragProperties ===
+// ======================
+
+/// This structure contains air dragging properties.
+#[derive(Default, Clone, Copy)]
+pub struct DragProperties {
+    coefficient: f32
+}
+
+impl DragProperties {
+    pub fn new(amount:f32) -> Self {
+        Self { coefficient: amount }
+    }
+}
+
+impl PhysicsForce for DragProperties {
+    fn force(&self, kinematics:&KinematicsProperties) -> Vector3<f32> {
+        let velocity  = kinematics.velocity;
+        let speed     = velocity.norm();
+        if speed > 0.0 {
+            velocity.normalize() * speed * speed * -0.5 * self.coefficient
+        } else {
+            zero()
+        }
+    }
+}
+
+// === Getters ===
+
+impl DragProperties {
+    pub fn coefficient(self) -> f32 {
+        self.coefficient
+    }
+}
+
+
+// === Setters ===
+
+impl DragProperties {
+    pub fn set_coefficient(&mut self, coefficient:f32) {
+        self.coefficient = coefficient
+    }
+}
+
+
+
+// ========================
+// === SpringProperties ===
+// ========================
+
+/// This structure contains spring physics properties.
+#[derive(Debug, Clone, Copy)]
+pub struct SpringProperties {
+    coefficient : f32,
+    fixed_point : Vector3<f32>
+}
+
+impl Default for SpringProperties {
+    fn default() -> Self {
+        Self::new(zero(),zero())
+    }
+}
```

Comment:
Why not deriving it? Here and other places

Reply:
It has been said a few times, Vector3 doesn't implement default.